public/app/features/plugins/admin/components/PluginDetailsPage.tsx

```diff
@@ -0,0 +1,112 @@
+import { css } from '@emotion/css';
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { GrafanaTheme2, NavModelItem } from '@grafana/data';
+import { useStyles2, TabContent, Alert } from '@grafana/ui';
+import { Layout } from '@grafana/ui/src/components/Layout/Layout';
+import { Page } from 'app/core/components/Page/Page';
+import { AppNotificationSeverity } from 'app/types';
+
+import { Loader } from '../components/Loader';
+import { PluginDetailsBody } from '../components/PluginDetailsBody';
+import { PluginDetailsDisabledError } from '../components/PluginDetailsDisabledError';
+import { PluginDetailsSignature } from '../components/PluginDetailsSignature';
+import { usePluginDetailsTabs } from '../hooks/usePluginDetailsTabs';
+import { usePluginPageExtensions } from '../hooks/usePluginPageExtensions';
+import { useGetSingle, useFetchStatus, useFetchDetailsStatus } from '../state/hooks';
+import { PluginTabIds } from '../types';
+
+export type Props = {
+  // The ID of the plugin
+  pluginId: string;
+  // The navigation ID used for displaying the sidebar navigation
+  navId?: string;
+  // Can be used to customise the title & subtitle for the not found page
+  notFoundNavModel?: NavModelItem;
+  // Can be used to customise the content shown when a plugin with the given ID cannot be found
+  notFoundComponent?: React.ReactElement;
+};
+
+export function PluginDetailsPage({
+  pluginId,
+  navId = 'plugins',
+  notFoundComponent = <NotFoundPlugin />,
+  notFoundNavModel = {
+    text: 'Unknown plugin',
+    subTitle: 'The requested ID does not belong to any plugin',
+    active: true,
+  },
+}: Props) {
+  const location = useLocation();
+  const queryParams = new URLSearchParams(location.search);
+  const plugin = useGetSingle(pluginId); // fetches the plugin settings for this Grafana instance
+  const { navModel, activePageId } = usePluginDetailsTabs(plugin, queryParams.get('page') as PluginTabIds);
+  const { actions, info, subtitle } = usePluginPageExtensions(plugin);
+  const { isLoading: isFetchLoading } = useFetchStatus();
+  const { isLoading: isFetchDetailsLoading } = useFetchDetailsStatus();
+  const styles = useStyles2(getStyles);
+
+  if (isFetchLoading || isFetchDetailsLoading) {
+    return (
+      <Page
+        navId={navId}
+        pageNav={{
+          text: 'Loading details...',
```

Comment:
nit: this text feels a bit intrusive in the header & breadcrumbs. IMO the Loader component is enough to signal to the user that something's still loading, and we could simply use `navModel here`.

Reply:
Or just an empty string? (Because we might hit a NotFoundError case after loading)
WDYT?

Reply:
Yeah, maybe you are right, I will try the empty strings for the title and subTitle. 👍 

(Why I didn't want to leave it empty is because then we have a flickering effect: even though there are no changes in the breadcrumbs, it displays the `"Plugins"` head title. Will try to record how it looks like initially.)

Reply:
I think it's better indeed.


https://user-images.githubusercontent.com/9974811/204517168-8e87ed83-4f7c-47ce-b969-ab6f8cb8da28.mov


Updated in https://github.com/grafana/grafana/pull/58741/commits/30ea6b1142180aa9518e2d8cdca0bb4999221e77.